app/helpers/application_helper.rb

```diff
@@ -464,4 +464,13 @@ def bracket_small_population(value, mask: true)
       bracket.first
     end
   end
+
+  # Determine whether the given search term is possibly a Primary Key (it's numeric and less than 4 bytes)
+  def possibly_pk?(search_term) # could add optional arg for 4 byte vs 8 byte, if needed later
+    numeric = /[\d-]+/.match(search_term).try(:[], 0) == search_term
+    return false unless numeric
+
+    max_pk = 2_147_483_648 # PK is a 4 byte signed INT (2 ** ((4 * 8) - 1))
+    search_term.to_i < max_pk
+  end
```

Comment:
Maybe this should go on `ApplicationRecord` as a class method?  Then you could call it:
```
ApplictaionRecord.possibly_pk?(1234)
```
wherever you need it.  That also means any class that inherits from ApplicationRecord would have the method, so enrollment (for instance) could just call `possibly_pk?(1234)` in the context of a  scope or class method.

ApplicationHelper is really intended for use in views.